ssm/ssm2.py

```diff
@@ -91,10 +99,29 @@ def __init__(self, hosts_and_ports, qpath, cert, key, dest=None, listen=None,
         
         # create the filesystem queues for accepted and rejected messages
         if dest is not None and listen is None:
-            self._outq = QueueSimple(qpath)
+            # Determine what sort of outgoing structure to make
+            if path_type == 'dirq':
+                if QueueSimple is None:
+                    raise ImportError("dirq path_type requested but the dirq "
+                                      "module wasn't found.")
+
+                self._outq = QueueSimple(qpath)
+
+            elif path_type == 'directory':
+                self._outq = MessageDirectory(qpath)
+            else:
+                raise Ssm2Exception('Unsupported path_type variable.')
+
         elif listen is not None:
             inqpath = os.path.join(qpath, 'incoming')
             rejectqpath = os.path.join(qpath, 'reject')
+
```

Comment:
Should check that `path_type == 'dirq'` and raise a warning if it's not and say something like the setting is wrong but is being ignored.

Reply:
This check seems like it would be a good idea.

`bin/receiver.py` doesn't pass `path_type` through to the SSM at the moment. So I could make it pass it through and put the check in `ssm/ssm2.py` or make the check in `bin/receiver.py`, thoughts?

Reply:
You don't need to do any of that. `path_type` has `dirq` as the default in the `__init__` of `Ssm2`.

Reply:
but the default `path_type` currently doesn't get changed by `bin/receiver.py` - even if it's set it in `conf/receiver.cfg`,

Reply:
Ok, we'll not worry about this then.